pkg/rbd/rbd_util.go

```diff
@@ -336,7 +338,66 @@ func getMonsAndClusterID(options map[string]string) (monitors, clusterID string,
 	return
 }
 
-func genVolFromVolumeOptions(volOptions map[string]string, disableInUseChecks bool) (*rbdVolume, error) {
+// isLegacyVolumeID checks if passed in volume ID string conforms to volume ID naming scheme used
+// by the version 1.0.0 (legacy) of the plugin, and returns true if found to be conforming
+func isLegacyVolumeID(volumeID string) bool {
+	// Version 1.0.0 volumeID format: "csi-rbd-vol-" + UUID string
+	//    length: 12 ("csi-rbd-vol-") + 36 (UUID string)
+
+	// length check
+	if len(volumeID) != 48 {
+		return false
+	}
+
+	// Header check
+	if !strings.HasPrefix(volumeID, "csi-rbd-vol-") {
+		return false
+	}
+
+	// Trailer UUID format check
+	if uuid.Parse(volumeID[12:]) == nil {
+		return false
+	}
+
+	return true
+}
+
+// upadateMons function is used to update the rbdVolume.Monitors for volumes that were provisioned
+// using the 1.0.0 version (legacy) of the plugin.
+func updateMons(rbdVol *rbdVolume, options, credentials map[string]string) error {
```

Comment:
same here.

Reply:
CephFS code is slightly different in the way it is written to extract code from options and from secrets. Either can be made to align with the other.

With the changes to CephFS using ceph Manager commands, should we do this now? As that commit is leaving out backward compat to be added again later afaict.

Reply:
we can do  it in while addressing backward compatibility in ceph manager commands